### PR TITLE
Update SWE and SRE job postings

### PR DESCRIPTION
**Pull request by aarongable**
Update the Senior Software Engineer job posting based on my
feedback from having gone through the hiring process with this
same posting recently. In addition, re-open the Site Reliability
Engineer posting.

### CODE REVIEW

**Comment by aarongable** [comment]
Notes:
* This is very much a draft, I'd love to have discussion on phrasing choices here.
* I've resurrected an old SRE job posting, but made only minor edits. Folks from the SRE side should definitely have input there and potentially re-draft the whole thing.
* I don't have permission to add reviewers/assignees in this repository, so I'm just going to send a link to this PR to some relevant folks and hope y'all take a look.

**Comment by bdaehlie** [comment]
The changes to the Software Engineer position look good to me. Please do not post an SRE position right now. If you can remove the SRE changes here we can merge the SE changes. Thanks!

File: content/en/careers/le-sr-sw-eng.html [hunk 1]
```diff
@@ -24,34 +24,36 @@
 <p>We’re looking for an additional software engineer. Maybe that’s you! Here’s what our software engineering team does:</p>
 
 <ul>
-	<li>We maintain Boulder, the server software that runs our ACME API. This includes:</li>
+	<li>We maintain <a href="https://github.com/letsencrypt/boulder">Boulder</a>, the server software that runs our Automated Certificate Management Environment (ACME) API. This includes:</li>
```
Comment (tgeoghegan) [inline content/en/careers/le-sr-sw-eng.html:27]: nit: add a hyperlink to RFC 8555, or the Wikipedia page for ACME? https://en.wikipedia.org/wiki/Automated_Certificate_Management_Environment

File: content/en/careers/le-sr-sw-eng.html [hunk 2]
```diff
@@ -24,34 +24,36 @@
...
 		</ul>
-	<li>We maintain Pebble, a “small size” version of Boulder that helps developers test their ACME clients.</li>
+	<li>We maintain <a href="https://github.com/letsencrypt/pebble">Pebble</a>, a “small size” version of Boulder that helps developers test their ACME clients.</li>
 	<li>We contribute to open source projects we depend on.</li>
 	<li>We help run a community forum that helps developers and end users fix their problems, and moderate comments to ensure everything stays on-track. We are happy to say ours is the friendliest and most supportive HTTPS-related forum around.</li>
-	<li>We participate in an on call rotation. We value protecting our teammates from burnout and accomodate people's schedules as needed.</li>
-	<li>We develop expertise on Public Key Infrastructure and advise the rest of the team on related changes.</li>
-	<li>We participate in standards-development processes at IETF and elsewhere.</li>
+	<li>We participate in an on call rotation, with infrequent pages and no strict service level thanks to our wonderful SRE colleagues. We value protecting our teammates from burnout and accomodate people's schedules as needed.</li>
```
Comment (tgeoghegan) [inline content/en/careers/le-sr-sw-eng.html:36]: nit: 2 ms in accommodate. Though that was wrong before your PR, too.

File: content/en/careers/le-sr-sw-eng.html (hunk 1 above)
Comment (aarongable) [reply content/en/careers/le-sr-sw-eng.html]: Good idea, done.

**Comment by aarongable** [comment]
Updated to address comments and remove link to the SRE listing. PTAL.

**Pull request merged**